task6/index.js

```diff
@@ -0,0 +1,364 @@
+                }
+                if (item === 'vendor') {
+                    elem = elem.filter(item => item.vendor === filterConfig.vendor);
+                }
+                else {
+                    return 'Error'
+                }
+            }
+        }
+        elem.sort(function (a, b) {
+            return a.createdAt - b.createdAt;
+        });
+        return elem.slice(skip, skip + top);
+    }
+
+    console.log(getAds(0, 10))
+    console.log(getAds(10, 10))
+    console.log(getAds('3'))
+    console.log(getAds(0, 4, {vendor: 'Анна Петрачкова'}))
+    console.log(getAds(0, 7, {validUntil: new Date('2021-05-01T24:00:00')}))
+
+
+    function getAd(id) {
+        return adList.filter(item => item.id == id)
+    }
+    console.log(getAd(12))
+
+    function addAd(newElem) {
+        if (validateAd(newElem)) {
+            for (let item of adList) {
+                if (item.id == newElem.id) {
+                    return 'Error'
+                }
+            }
+            adList.push(newElem)
+            return adList
+        }
+        else return false
+    }
+    console.log(addAd({id: '21',
+        description: 'Суши от Rollbox',
+        createdAt: new Date('2021-03-01T11:00:00'),
+        link: 'https://rollbox.by/',
+        vendor: 'Rollbox',
+        photoLink: 'https://sushichefarts.by/upload/iblock/ba7/ba79b531e47dfa1bec62fce7d1817916.jpg',
+        validUntil: new Date('2021-07-01T24:00:00'),
+        discount: '40%',
+        hashTags: ['Прочие услуги']}))
+
+    console.log(addAd({id: '14',
+        description: 'Суши от Rollbox',
+        createdAt: new Date('2021-03-01T11:00:00'),
+        link: 'https://rollbox.by/',
+        vendor: 'Rollbox',
+        photoLink: 'https://sushichefarts.by/upload/iblock/ba7/ba79b531e47dfa1bec62fce7d1817916.jpg',
+        validUntil: new Date('2021-07-01T24:00:00'),
+        discount: '40%',
+        hashTags: ['Прочие услуги']}))
+
+    function editAd(id, propertyChange, changedValue) {
+        if (propertyChange === 'id' || propertyChange === 'vendor' || propertyChange === 'createdAt') {
+            return 'Error'
+        }
+        for (let item of adList) {
+            if (item.id === id) {
+                for (let key in item) {
+                    if (key === propertyChange) {
+                        item[key] = changedValue
+                    }
+                }
+            }
+        }
+        return adList
+    }
+    console.log(editAd(4, 'discount', '70%'))
+    console.log(editAd(16, 'id', '23'))
+
+
+    function validateAd(elem) {
+        return typeof elem.id == 'string' && elem.id !== ""
+            && typeof elem.description == 'string' && elem.description.length <= 200 && typeof elem.link == 'string' && elem.description !== ""
+            && elem.createdAt instanceof Date && elem.createdAt !== ""
+            && typeof elem.link == 'string' && elem.link !== ""
```

Comment:
I'd simplify this conditions to `typeof elem.link == 'string' && elem.link`, that's equivalent.